packages/rocketchat-slackbridge/slackbridge.js

```diff
@@ -0,0 +1,712 @@
+/* globals logger */
+
+class SlackBridge {
+	constructor() {
+		this.slackClient = Npm.require('slack-client');
+		this.apiToken = RocketChat.settings.get('SlackBridge_APIToken');
+		this.rtm = {};
+		this.connected = false;
+		this.userTags = {};
+
+		RocketChat.settings.onload('SlackBridge_APIToken', (key, value) => {
+			this.apiToken = value;
+			if (this.connected) {
+				this.disconnect();
+				this.connect();
+			} else if (RocketChat.settings.get('SlackBridge_Enabled')) {
+				this.connect();
+			}
+		});
+
+		RocketChat.settings.onload('SlackBridge_Enabled', (key, value) => {
+			if (value && this.apiToken) {
+				this.connect();
+			} else {
+				this.disconnect();
+			}
+		});
+	}
+
+	connect() {
+		if (this.connected === false) {
+			this.connected = true;
+			logger.connection.info('Connecting via token: ', this.apiToken);
+			var RtmClient = this.slackClient.RtmClient;
+			this.rtm = new RtmClient(this.apiToken);
+			this.rtm.start();
+			this.setEvents();
+		}
+	}
+
+	disconnect() {
+		if (this.connected === true) {
+			this.rtm.disconnect && this.rtm.disconnect();
+			this.connected = false;
+			logger.connection.info('Disconnected');
+		}
+	}
+
+	convertSlackMessageToRocketChat(message) {
+		if (!_.isEmpty(message)) {
+			message = message.replace(/<!everyone>/g, '@all');
+			message = message.replace(/<!channel>/g, '@all');
+			message = message.replace(/&gt;/g, '<');
+			message = message.replace(/&lt;/g, '>');
+			message = message.replace(/&amp;/g, '&');
+			message = message.replace(/:simple_smile:/g, ':smile:');
+			message = message.replace(/:memo:/g, ':pencil:');
+			message = message.replace(/:piggy:/g, ':pig:');
+			message = message.replace(/:uk:/g, ':gb:');
+			message = message.replace(/<(http[s]?:[^>]*)>/g, '$1');
+
+			message.replace(/(?:<@)([a-zA-Z0-9]+)(?:\|.+)?(?:>)/g, (match, userId) => {
+				if (!this.userTags[userId]) {
+					this.findUser(userId) || this.addUser(userId); // This adds userTags for the userId
+				}
+				let userTags = this.userTags[userId];
+				if (userTags) {
+					message = message.replace(userTags.slack, userTags.rocket);
+				}
+			});
+		} else {
+			message = '';
+		}
+		return message;
+	}
+
+	findChannel(channelId) {
+		return RocketChat.models.Rooms.findOneByImportId(channelId);
+	}
+
+	addChannel(channelId, hasRetried = false) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (18)](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2677595092/issues/source?bid=3199634&fileBranchId=3334129#l81)